lib/cell/concept.rb

```diff
@@ -5,8 +5,23 @@ class Concept < Cell::ViewModel
 
     # TODO: this should be in Helper or something. this should be the only entry point from controller/view.
     class << self
-      def class_from_cell_name(name)
-        name.camelize.constantize
+      def class_from_cell_name(name, controller)
+        @cell_class ||= begin
+          supports_path = controller.supports_path?
+          routes  = controller.respond_to?(:_routes)  && controller._routes
+          helpers = controller.respond_to?(:_helpers) && controller._helpers
+
+          Class.new(name.camelize.constantize) do
```

Comment:
> you misunderstanding of Cells and OOP in general

A rather sweeping assumption. The framework itself (Rails), like all frameworks, utilizes multiple paradigms to begin with, not just OO. Each paradigm solves a problem domain. OO does not address the issue of cross cutting concerns, which is addressed by AO. 

This isn't a question of misunderstanding OOP, nor Cells. This is a question of purity of ideology standing in the way of practicality. The fact that there's no alternative provided, optional or otherwise is misguided. Again, if the "pure OO design" forces your users to circumvent it, what is the gain?

Yes, "helpers are shit", I read the memo. But without them Rails is useless with or without cells without extensive adjustments. If the framework is actually supported, then the whole framework should be supported. It makes no sense to just ignore the ugly bits and at the same time actively block any attempt to provide proper access to the underlying framework.